File: BlackJack/BlackjackModel/Player.py
from Card import *

#One of the blackjack players

class Player:
    def __init__(self):
        self.__hand = []
        self.__gameState = "playing" #Keeps players from playing out of turn
        
    def hand(self):
        return self.__hand.copy()
        
    def gameState(self):
        return self.__gameState
    
    def setGameState(self, state):
        self.__gameState = state
        
    #Add a card to the hand
    def hit(self, card):
        if self.__gameState == "playing":
            self.__hand.append(card)
            
    #Sum up the cards in the hand
    def countHand(self):
        total = 0
        aceCount = 0 #Keep a count of the number of aces in the hand
        
        for card in self.__hand:
            #The  ace  has  the  value  of  11...
            if card is Card.ACE:
                total += 11
                aceCount += 1
                
            #Jack, Queen, and King are all worth 10.
            elif card is Card.JACK or card is Card.QUEEN or card is Card.KING:
                total += 10
                
            else:
                total += card.value
            
        #...Unless it is in a hand with a value over 21 in which case it has the value of 1.
        while aceCount > 0 and total > 21:
            total -= 10
            aceCount -= 1
            
        return total
            

File: BlackJack Java/Player.java
package BlackjackModel;

/*One of the blackjack players*/

import java.util.*;

public class Player {
    private List<Card> hand;
    private BlackjackState gameState; //To determine when their turn is
    private int score; //Total number of games won. This isn't necessarily needed, but might as well include it.
    
    public Player() {
        hand = new LinkedList<Card>();
        gameState = BlackjackState.PLAYING;
        score = 0;
    }
    
    //Getters
    public List<Card> hand() { return hand; }
    public BlackjackState gameState() { return gameState; }
    public int totalScore() { return score; }
    
    public void setGameState(BlackjackState state) {
        if(state.equals(BlackjackState.WON))
            score++;
        
        gameState = state;
    }
    
    public void hit(Card c) {
        if(gameState.equals(BlackjackState.PLAYING))
            hand.add(c);
    }
    
    public int countHand() {
        int total = 0;
        int aceCount = 0;
        
        for(Card c : hand) {
            switch(c) {
                //Keep a count of the number of aces in the hand
                case ACE: {
                    total += 11; 
                    aceCount++; 
                    break;
                }
                case TWO: total += 2; break;
                case THREE: total += 3; break;
                case FOUR: total += 4; break;
                case FIVE: total += 5; break;
                case SIX: total += 6; break;
                case SEVEN: total += 7; break;
                case EIGHT: total += 8; break;
                case NINE: total += 9; break;
                default: total += 10;
            }
        }
        
        //Count aces as 1, not 11 as needed.
        while(aceCount > 0 && total > 21) {
            total -= 10;
            aceCount--;
        }
        
        return total;
    }
}